kore/src/Kore/Attribute/UniqueId.hs

```diff
@@ -0,0 +1,88 @@
+{-|
+Module      : Kore.Attribute.UniqueId
+License     : NCSA
+
+The unique id attribute identifies symbols, sorts, axioms and
+various other things.
+
+Example:
+@UNIQUE'Unds'ID{}("07a34b11585162c291311c03441e08beb2532e48d4ece33b9d58a9456f2f7623")@
+-}
+module Kore.Attribute.UniqueId
+    ( UniqueId (..)
+    , uniqueIdId
+    , uniqueIdSymbol
+    , uniqueIdAttribute
+    ) where
+
+import Data.Text
+    ( Text
+    )
+import qualified Generics.SOP as SOP
+import qualified GHC.Generics as GHC
+
+import Kore.Attribute.Parser as AttributeParser
+import Kore.Debug
+import qualified Kore.Error
+    ( koreFail
+    )
+
+
+{- | @UniqueId@ represents the @uniqueId@ attribute for axioms.
+ -}
+newtype UniqueId = UniqueId { getUniqueId :: Maybe Text }
+    deriving (Eq, GHC.Generic, Ord, Show)
+
+instance SOP.Generic UniqueId
+
+instance SOP.HasDatatypeInfo UniqueId
+
+instance Debug UniqueId
+
+instance NFData UniqueId
+
+instance Default UniqueId where
+    def = UniqueId Nothing
+
+-- | Kore identifier representing the @uniqueId@ attribute symbol.
+uniqueIdId :: Id
+uniqueIdId = "UNIQUE'Unds'ID"
+
+-- | Kore symbol representing the @uniqueId@ attribute.
+uniqueIdSymbol :: SymbolOrAlias
+uniqueIdSymbol =
+    SymbolOrAlias
+        { symbolOrAliasConstructor = uniqueIdId
+        , symbolOrAliasParams = []
+        }
+
+-- | Kore pattern representing the @uniqueId@ attribute.
+uniqueIdAttribute :: Text -> AttributePattern
+uniqueIdAttribute uniqueId =
+    attributePattern uniqueIdSymbol [attributeString uniqueId]
+
+instance ParseAttributes UniqueId where
+    parseAttribute = withApplication' parseApplication
+      where
+        parseApplication
+            :: [Sort]
+            -> [AttributePattern]
+            -> UniqueId
+            -> AttributeParser.Parser UniqueId
+        parseApplication params args (UniqueId Nothing) = do
+            AttributeParser.getZeroParams params
+            case args of
+                [_] -> do
+                    arg <- AttributeParser.getOneArgument args
+                    StringLiteral str <- AttributeParser.getStringLiteral arg
+                    return (UniqueId (Just str))
+                _ -> Kore.Error.koreFail
```

Comment:
This condition should be detected, and an error thrown, by `AttributeParser.getOneArgument`.

Reply:
Fixed.